File: others/프로그래머스_HIndex.py
def solution(citations):
    # 최솟값이 논문 수 보다 크다면,
    # 논문 개수 그대로 반환
    if min(citations) >= len(citations):
        return len(citations)

    # 내림차순 정렬
    citations.sort(reverse=True)

    # 인용 많은 순으로 확인하면서 내려간다
    for (idx, citation) in enumerate(citations):
        if idx + 1 > citation:
            return idx

File: others/프로그래머스_HIndex.js
function solution(citations) {

  if (Math.min(...citations) >= citations.length) {
    return citations.length
  }

  citations.sort(function(a, b) {return b-a})

  for (let i=0; i < citations.length; i++) {
    if (i+1 > citations[i]) {
      return i
    }
  }
}

console.log(solution([3, 0, 6, 1, 5]))